src/ucs/memory/rcache_vfs.c

```diff
@@ -0,0 +1,134 @@
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <ucs/config/global_opts.h>
+#include <ucs/datastruct/queue.h>
+#include <ucs/sys/compiler.h>
+#include <ucs/sys/math.h>
+#include <ucs/sys/string.h>
+#include <ucs/vfs/base/vfs_cb.h>
+#include <ucs/vfs/base/vfs_obj.h>
+#include "rcache_int.h"
+#include "rcache_vfs.h"
+
+
+#define UCS_RCACHE_FORMAT_STRING ("%12s:%9zi\n")
+
+
+static void ucs_rcache_vfs_read_inv_q_length(void *obj,
+                                             ucs_string_buffer_t *strb,
+                                             void *arg_ptr, uint64_t arg_u64)
+{
+    ucs_rcache_t *rcache = obj;
+    size_t rcache_inv_q_length;
+
+    ucs_spin_lock(&rcache->lock);
+    rcache_inv_q_length = ucs_queue_length(&rcache->inv_q);
+    ucs_spin_unlock(&rcache->lock);
+
+    ucs_string_buffer_appendf(strb, "%zu\n", rcache_inv_q_length);
+}
+
+static void ucs_rcache_vfs_read_gc_list_length(void *obj,
+                                               ucs_string_buffer_t *strb,
+                                               void *arg_ptr, uint64_t arg_u64)
+{
+    ucs_rcache_t *rcache = obj;
+    unsigned long rcache_gc_list_length;
+
+    ucs_spin_lock(&rcache->lock);
+    rcache_gc_list_length = ucs_list_length(&rcache->gc_list);
+    ucs_spin_unlock(&rcache->lock);
+
+    ucs_string_buffer_appendf(strb, "%lu\n", rcache_gc_list_length);
+}
+
+static void ucs_rcache_vfs_calc_regions_distribution(ucs_rcache_t *rcache,
+                                                     size_t *rcache_dist,
+                                                     size_t num_bins)
+{
+    ucs_config_thresholds_t *thresholds = &ucs_global_opts.rcache_thresholds;
+    ucs_rcache_region_t *region;
+    size_t region_size, i;
+
+    for (i = 0; i < num_bins; ++i) {
+        rcache_dist[i] = 0;
+    }
+
+    ucs_spin_lock(&rcache->lru.lock);
+
+    ucs_list_for_each(region, &rcache->lru.list, lru_list) {
+        region_size = region->super.end - region->super.start;
+
+        for (i = 0; i < num_bins - 1; ++i) {
+            if (region_size < thresholds->vals[i]) {
+                break;
+            }
+        }
+        ++rcache_dist[i];
+    }
+
+    ucs_spin_unlock(&rcache->lru.lock);
+}
+
+static void ucs_rcache_vfs_dump_regions_distribution(const size_t *rcache_dist,
+                                                     size_t num_bins,
+                                                     ucs_string_buffer_t *strb)
+{
+    ucs_config_thresholds_t *thresholds = &ucs_global_opts.rcache_thresholds;
+    size_t i, start, end;
+    char range_str[64];
+
+    for (i = 0; i < num_bins; ++i) {
+        start = (i != 0) ? thresholds->vals[i - 1] : 0;
+        end   = (i != num_bins - 1) ? thresholds->vals[i] : UCS_MEMUNITS_INF;
+
+        ucs_memunits_range_str(start, end, range_str, sizeof(range_str));
+        ucs_string_buffer_appendf(strb, UCS_RCACHE_FORMAT_STRING, range_str,
```

Comment:
maybe we can do something like this path:
ucs/rcache/dist/[size]/count
where [size] is the maximal size of the bin (power-of-2)
I prefer the VFS output to be  more "raw" or "low-level" so it could be easier used in scripts/grep etc.